src/analysis/convex/basic.lean

```diff
@@ -254,6 +255,71 @@ begin
   simp [set.mem_image, mem_add, eq_comm]
 end
 
+lemma convex.combo_to_vadd {a b : ℝ} {x y : E} (h : a + b = 1) :
+  a • x + b • y = b • (y - x) + x  :=
+  eq.symm (calc
+    b • (y - x) + x = b • (y - x) + (1 : ℝ) • x          : by rw [one_smul]
+                ... = b • (y - x) + (a + b) • x           : by rw [h]
+                ... = (b • y - b • x) + (a • x + b • x)   : by rw [add_smul, smul_sub]
+                ... = a • x + b • y                       : by abel )
+
+/--
+Applying an affine map to an affine combination of two points yields
+an affine combination of the images.
+ -/
+lemma convex.combo_affine_apply {a b : ℝ} {x y : E} {f : affine_map ℝ E E F F}
+  (h : a + b = 1) : f (a • x + b • y) = a • (f x) + b • (f y) :=
+begin
+  rw [convex.combo_to_vadd h],
+  calc
+    f (b • (y - x) + x)
+          = f.to_fun (b • ((y -ᵥ x) : E) +ᵥ x)                    : rfl
+      ... = f.linear.to_fun (b • ((y -ᵥ x) : E)) +ᵥ f.to_fun x    : by rw [f.map_vadd']; refl
+      ... = b • f.linear ((y -ᵥ x) : E) + f.to_fun x              : by rw [f.linear.map_smul']; refl
+      ... = b • (f y - f x) + f x                       : by rw [affine_map.linear_map_vsub]; refl
+      ... = b • (f y) - b • (f x) + f x                 : by rw [smul_sub]
+      ... = b • f y - b • f x + (1 : ℝ) • f x           : by rw [one_smul]
+      ... = b • f y - b • f x + (a + b) • f x           : by rw [h]
+      ... = b • f y - b • f x + (a • f x + b • f x)     : by rw [add_smul]
+      ... = a • f x + b • f y                           : by abel
+end
+
+/-- Convexity is preserved by affine maps -/
+lemma convex.affine_preimage {f : affine_map ℝ E E F F} {s : set F} (hs : convex s) :
+  convex (f ⁻¹' s) :=
+begin
+  intros x y xs ys a b ha hb hab,
+  rw [mem_preimage, convex.combo_affine_apply hab],
+  exact hs xs ys ha hb hab,
+end
+
+/-- Convexity is preserved by affine maps -/
+lemma convex.affine_image {f : affine_map ℝ E E F F} {s : set E} (hs : convex s) :
+  convex (f '' s) :=
+begin
+  intros x y xs ys a b ha hb hab,
+  rcases xs with ⟨x', ⟨hx', hx'f⟩⟩,
+  rcases ys with ⟨y', ⟨hy', hy'f⟩⟩,
+  refine ⟨a • x' + b • y',⟨hs hx' hy' ha hb hab,_⟩⟩,
+  rw [convex.combo_affine_apply, hx'f, hy'f],
+  exact hab,
+end
+
+/-- The translation of a convex set is also convex -/
+lemma convex.translate_preimage_right (hs : convex s) (a : E) : convex ((λ z, a + z) ⁻¹' s) :=
+begin
+  let g : affine_map ℝ E E E E := ⟨λ z, a + z, id, by intros p v; dsimp; abel⟩,
+  change convex (g ⁻¹' s),
```

Comment:
Does the proof work without `change`? (Same question for the other occurrences below.)

Reply:
No, they don't work without `change`.